imports/plugins/included/taxes-avalara/client/containers/avalaraSettingsFormContainer.js

```diff
@@ -0,0 +1,153 @@
+      return;
+    }
+    Meteor.call("package/update", "taxes-avalara", "settings", changedInfo.doc.settings, (error) => {
+      if (error) {
+        Alerts.toast(
+          i18next.t("admin.update.avalaraUpdateFailed", { defaultValue: "Failed to update Avalara settings." }),
+          "error"
+        );
+        return;
+      }
+      Alerts.toast(
+        i18next.t("admin.update.avalaraUpdateSucceeded", { defaultValue: "Avalara settings updated." }),
+        "success"
+      );
+    });
+  },
+
+  handleTestCredentials(event, settings) {
+    Meteor.call("avalara/testCredentials", settings.avalara, function (error, result) {
+      if (error && error.message) {
+        return Alerts.toast(`${i18next.t("settings.testCredentialsFailed")} ${error.message}`, "error");
+      }
+      try {
+        const statusCode = _.get(result, "statusCode");
+        const connectionValid = _.inRange(statusCode, 400);
+        if (connectionValid) {
+          return Alerts.toast(i18next.t("settings.testCredentialsSuccess"), "success");
+        }
+        return Alerts.toast(i18next.t("settings.testCredentialsFailed"), "error");
+      } catch (e) {
+        return Alerts.toast(i18next.t("settings.testCredentialsFailed"), "error");
+      }
+    });
+  }
+};
+
+const countryDefaults = ["US", "CA"];
+let validCountries = null;
+
+const composer = (props, onData) => {
+  const shownFields = {
+    ["settings.avalara.apiLoginId"]: AvalaraPackageConfig._schema["settings.avalara.apiLoginId"],
+    ["settings.avalara.username"]: AvalaraPackageConfig._schema["settings.avalara.username"],
+    ["settings.avalara.password"]: AvalaraPackageConfig._schema["settings.avalara.password"],
+    ["settings.avalara.companyCode"]: AvalaraPackageConfig._schema["settings.avalara.companyCode"],
+    ["settings.avalara.shippingTaxCode"]: AvalaraPackageConfig._schema["settings.avalara.shippingTaxCode"],
+    ["settings.addressValidation.enabled"]: AvalaraPackageConfig._schema["settings.addressValidation.enabled"],
+    ["settings.addressValidation.countryList"]: AvalaraPackageConfig._schema["settings.addressValidation.countryList"],
+    ["settings.avalara.requestTimeout"]: AvalaraPackageConfig._schema["settings.avalara.requestTimeout"],
+    ["settings.avalara.mode"]: AvalaraPackageConfig._schema["settings.avalara.mode"],
+    ["settings.avalara.performTaxCalculation"]: AvalaraPackageConfig._schema["settings.avalara.performTaxCalculation"],
+    ["settings.avalara.enableLogging"]: AvalaraPackageConfig._schema["settings.avalara.enableLogging"],
+    ["settings.avalara.logRetentionDuration"]: AvalaraPackageConfig._schema["settings.avalara.logRetentionDuration"],
+    ["settings.avalara.commitDocuments"]: AvalaraPackageConfig._schema["settings.avalara.commitDocuments"]
+  };
+  const hiddenFields = [
+    "settings.avalara.enabled",
+    "settings.avalara.companyId"
+  ];
+
+  validCountries = validCountries || Countries.find({ value: { $in: countryDefaults } }).fetch();
```

Comment:
A `find` in minimongo is pretty much the same as reading the local variable, so I would not bother with caching the result in `validCountries`.